services/src/main/java/org/keycloak/services/clientpolicy/executor/SecureRequestObjectExecutor.java

```diff
@@ -0,0 +1,508 @@
+
+        public String getScope() {
+            return scope;
+        }
+
+        public void setScope(String scope) {
+            this.scope = scope;
+        }
+
+        public String getLoginHint() {
+            return loginHint;
+        }
+
+        public void setLoginHint(String loginHint) {
+            this.loginHint = loginHint;
+        }
+
+        public String getPrompt() {
+            return prompt;
+        }
+
+        public void setPrompt(String prompt) {
+            this.prompt = prompt;
+        }
+
+        public String getNonce() {
+            return nonce;
+        }
+
+        public void getNonce(String nonce) {
+            this.nonce = nonce;
+        }
+
+        public Integer getMax_age() {
+            return max_age;
+        }
+
+        public void setMax_age(Integer max_age) {
+            this.max_age = max_age;
+        }
+
+        public String getUiLocales() {
+            return uiLocales;
+        }
+
+        public void setUiLocales(String uiLocales) {
+            this.uiLocales = uiLocales;
+        }
+
+        public JsonNode getClaims() {
+            return claims;
+        }
+
+        public void setClaims(JsonNode claims) {
+            this.claims = claims;
+        }
+
+        public String getAcr() {
+            return acr;
+        }
+
+        public void setAcr(String acr) {
+            this.acr = acr;
+        }
+
+        public String getCodeChallenge() {
+            return codeChallenge;
+        }
+
+        public void setCodeChallenge(String codeChallenge) {
+            this.codeChallenge = codeChallenge;
+        }
+
+        public String getCodeChallengeMethod() {
+            return codeChallengeMethod;
+        }
+
+        public void setCodeChallengeMethod(String codeChallengeMethod) {
+            this.codeChallengeMethod = codeChallengeMethod;
+        }
+
+        public String getDisplay() {
+            return display;
+        }
+
+        public void setDisplay(String display) {
+            this.display = display;
+        }
+
+        public String getIdpHint() {
+            return idpHint;
+        }
+
+        public void setIdpHint(String idpHint) {
+            this.idpHint = idpHint;
+        }
+
+        public String getAction() {
+            return action;
+        }
+
+        public void setAction(String action) {
+            this.action = action;
+        }
+
+    }
+
+    @Override
+    public String getName() {
+        return componentModel.getName();
+    }
+
+    @Override
+    public String getProviderId() {
+        return componentModel.getProviderId();
+    }
+
+    private void DumpQueryParameters() {
+        MultivaluedMap<String, String> params = session.getContext().getUri().getQueryParameters();
+        params.keySet().forEach(s->ClientPolicyLogger.logv(logger, "Query Parameter : {0} = {1}", s, params.getFirst(s)));
+    }
+
+    private void DumpRequestObject() {
+        if (parsedRequestObject.getId() != null) 
```

Comment:
If we move most of the claims to the "otherClaims" and not explicitly contain them as properties of the AuthorizationEndpointRequestObject  (See my comment above), then this algorithm can be perhaps easier without explicitly naming every property. Just dumping the "otherClaims" map will do the thing.

Reply:
@mposolda Instead of using dedicated `AuthorizationEndpointRequestObject`, I'll use bearer `JsonNode` for parsedRequestObject.